pkg/daemon/ceph/osd/kms/volumes.go

```diff
@@ -82,25 +82,36 @@ func VaultVolumeAndMount(config map[string]string) (v1.Volume, v1.VolumeMount) {
 func tlsSecretPath(tlsOption string) string {
 	switch tlsOption {
 	case api.EnvVaultCACert:
-		return vaultCAFileName
+		return VaultCAFileName
 	case api.EnvVaultClientCert:
-		return vaultCertFileName
+		return VaultCertFileName
 	case api.EnvVaultClientKey:
-		return vaultKeyFileName
+		return VaultKeyFileName
 
 	}
 
 	return ""
 }
 
 // VaultTokenFileVolume save token from secret as volume mount
-func VaultTokenFileVolume(tokenSecretName string) v1.Volume {
-	return v1.Volume{
+func VaultTokenFileVolume(tokenSecretName string, config map[string]string) v1.Volume {
+	mode := int32(0400)
+	secretVolumeProjections := TLSSecretVolumeAndMount(config)
+
+	if tokenSecretName != "" {
+		projectionSecret := &v1.SecretProjection{Items: []v1.KeyToPath{{Key: KMSTokenSecretNameKey, Path: VaultFileName, Mode: &mode}}}
+		projectionSecret.Name = tokenSecretName
+		secretProjection := v1.VolumeProjection{Secret: projectionSecret}
+		secretVolumeProjections = append(secretVolumeProjections, secretProjection)
+	}
+	volume := v1.Volume{
 		Name: secrets.TypeVault,
 		VolumeSource: v1.VolumeSource{
-			Secret: &v1.SecretVolumeSource{
-				SecretName: tokenSecretName,
-				Items: []v1.KeyToPath{
-					{Key: KMSTokenSecretNameKey, Path: VaultFileName},
-				}}}}
+			Projected: &v1.ProjectedVolumeSource{
+				Sources: secretVolumeProjections,
+			},
+		},
+	}
+
+	return volume
 }
```

Comment:
In the interest of code sharing, can `VaultVolumeAndMount` be reused here and `VaultTokenFileVolume` be removed? Thoughts @leseb ?